src/dotnet/CarbonAware.WebApi/Controllers/CarbonAwareController.cs

```diff
@@ -56,12 +63,16 @@ public async Task<IActionResult> GetEmissionsDataForLocationsByTime([FromQuery(N
     [HttpGet("bylocation")]
     public async Task<IActionResult> GetEmissionsDataForLocationByTime(string location, DateTime? time = null, DateTime? toTime = null, int durationMinutes = 0)
     {
-        var props = new Dictionary<string, object?>() {
+        var props = new Dictionary<string, object>() {
             { CarbonAwareConstants.LOCATIONS, new List<string>(){ location } },
             { CarbonAwareConstants.START, time ?? DateTime.Now },
-            { CarbonAwareConstants.END, toTime },
             { CarbonAwareConstants.DURATION, durationMinutes },
         };
+        
+        if(toTime != null) 
```

Comment:
Note: did this to avoid adding null values to the dictionary. Eventually I imagine we would parse the param string passed in and only include the fields that are specified (and so by default all non-null)

Reply:
> Note: we did this to avoid adding null values to the dictionary. Eventually I imagine we would parse the param string passed in and only include the fields that are specified (and so by default all non-null)

Each layer are responsible of handling the values passed. So my point is that it should handled on both sides.